### ISSUE TITLE
Accessibility for Search in API Reference

### ISSUE THREAD

**Issue by kayla-beckham** [scalar/scalar#3421]
### What happens?

![Screen Shot 2024-10-09 at 4 00 24 PM](https://github.com/user-attachments/assets/bebd8020-543d-4bb2-bbe5-374bf853a02f)
![Screen Shot 2024-10-09 at 4 01 11 PM](https://github.com/user-attachments/assets/2c95a33f-06c8-48dd-9e1d-ec48bed32126)
1. No visible focus for Search input in navigation sidebar. When using keyboard for navigation, visible focus skips over the search box to the sidebar heading link title.
    1. In the search dialog pop up, the search box does not get a focus box, but it is possible to use the keyboard to navigate to the search input. The typing line shows up, but a well-defined (highly visible) visual indication of keyboard focus needs to be provided for each active element. 
2. The search dialog should have role=‘dialog’ or it should be wrapped in a dialog element.
    1. The search dialog should have an aria-label that is announced by a screen reader when it is displayed
3. When search dialog is opened, the navigation instructions in the footer of the dialog should be announced by a screen reader
4. When the search dialog is open and the user presses enter on the empty search field, the dialog closes. Is that expected behavior? If there is text in the search field and you press enter, the dialog remains open and shows search results.
5. After searching in the dialog, screen reader does not announce results of the search. It should announce the number of results found or the message ‘no results found’

### What did you expect to happen?

When using keyboard navigation, focus should be able to reach all interactive elements. Focus highlights should be visible. Components should have the correct roles and aria attributes so that important visual elements can be read by a screen reader. The status of expand/collapse or search events should be announced by a screen reader.
